app/routes/messages.rb

```diff
@@ -0,0 +1,24 @@
+Mumukit::Platform.map_organization_routes!(self) do
+  post '/courses/:course/messages' do
+    authorize! :teacher
+    message = json_body[:message]
+    message[:sender] = current_user.uid
+    assignment = Assignment.find_by!(with_organization_and_course 'exercise.eid': json_body[:exercise_id], 'student.uid': json_body[:uid])
```

Comment:
This route is overly complex. Please delegate it in more high-level model methods